packages/google-cloud-serverless/test/gcpfunction/events.test.ts

```diff
@@ -45,22 +44,24 @@ describe('wrapEventFunction', () => {
 
   function handleEvent(fn: EventFunctionWithCallback): Promise<any> {
     return new Promise((resolve, reject) => {
-      // eslint-disable-next-line deprecation/deprecation
-      const d = domain.create();
       const context = {
         eventType: 'event.type',
         resource: 'some.resource',
       };
-      d.on('error', reject);
-      d.run(() =>
-        process.nextTick(fn, {}, context, (err: any, result: any) => {
-          if (err != null || err != undefined) {
-            reject(err);
-          } else {
-            resolve(result);
-          }
-        }),
-      );
+
+      (async () => {
```

Comment:
Do we even need to make this an IIF?

Reply:
not needed per se - it's here to simplify error handling with `await` to get an exception for both synchronous and asynchronous flows.

can be easily replaced with Promise instead. let me push an update

Reply:
Applied a change, please let me know if it looks better now